### PR TITLE
feat: expose vo & hwdec in VideoControllerConfiguration

### PR DESCRIPTION
**Pull request by alexmercerind**
Added a new `VideoControllerConfiguration` class to use in composition with `VideoController`.

Similar to `PlayerConfiguration` in `Player`; like we have it already. As we are moving ahead, people seem to require more specific things & this makes things bit more opinionated IMO.

~~Also:~~

~~Should `enableHardwareAcceleration` be moved to `VideoControllerConfiguration` as-well?~~
~~We should also give a thought to `width` & `height` then; because they are mutable (thanks to `setSize` API).~~

### CODE REVIEW

**Comment by birros** [comment]
Moving all the `VideoController` parameters to `VideoControllerConfiguration` would be consistent with the `Player` configuration.

Some suggestions:

Maybe take the opportunity to rename `enableHardwareAcceleration` to `hardwareAcceleration`, since it's a boolean.
Or an enum, to be future-proof, if we want to preview a new acceleration type, like mpv does with `--vo`. For example `PlatformView`, `MoltenVK`…

~~Concerning `width` and `height`, to distinguish their immutable nature, maybe add the prefix `initial` -> `initialWidth`, specifying in their docs that the value can be modified thanks to `setSize`.~~
I think that's a bit redundant, we know that if `setSize` is present in the api that means that these properties are mutable…

**Review by birros — APPROVED** [review]
Otherwise, everything is ok for me.

File: media_kit_video/windows/video_output.h [hunk 1]
```diff
@@ -25,6 +25,19 @@
 #include "angle_surface_manager.h"
 #include "thread_pool.h"
 
+typedef struct _VideoOutputConfiguration {
```
Comment (alexmercerind) [inline media_kit_video/windows/video_output.h:28]: ^^

File: media_kit_video/lib/src/video_controller/native_video_controller/real.dart [hunk 2]
```diff
@@ -84,9 +115,12 @@ class NativeVideoController extends PlatformVideoController {
       'VideoOutputManager.Create',
       {
         'handle': handle.toString(),
-        'width': width.toString(),
-        'height': height.toString(),
-        'enableHardwareAcceleration': enableHardwareAcceleration,
+        'configuration': {
```
[the remark this answers is not in the document]
Comment (birros) [reply media_kit_video/lib/src/video_controller/native_video_controller/real.dart:118]: I agree, I implement it immediately.

**Pull request merged**